plugins/dirlisting/dirlisting.c

```diff
@@ -39,7 +39,7 @@ const mk_ptr_t mk_dir_iov_none  = mk_ptr_init("");
 const mk_ptr_t mk_dir_iov_slash = mk_ptr_init("/");
 
 /* Function wrote by Max (Felipe Astroza), thanks! */
-static char *mk_dirhtml_human_readable_size(char *buf, size_t size, int len)
+static char *mk_dirhtml_human_readable_size(char *buf, size_t size, size_t len)
```

Comment:
Maybe rename _size_ to _buf_size_ to avoid this sort of problem.
And _len_ to _file_len_ or _file_size_ (look more english).